tools/ir-generator/irclass.cpp

```diff
@@ -131,16 +131,16 @@ void IrDefinitions::generate(std::ostream &t, std::ostream &out, std::ostream &i
         << "#include \"ir/namemap.h\"         // IWYU pragma: keep\n"
         << "#include \"ir/node.h\"            // IWYU pragma: keep\n"
         << "#include \"ir/nodemap.h\"         // IWYU pragma: keep\n"
+        << "#include \"ir/unpacker_table.h\"  // IWYU pragma: keep\n"
         << "#include \"ir/vector.h\"          // IWYU pragma: keep\n"
         << "#include \"lib/ordered_map.h\"    // IWYU pragma: keep\n"
         << std::endl
         << "namespace P4 {\n"
         << std::endl
         << "class JSONLoader;\n"
-        << "using NodeFactoryFn = IR::Node*(*)(JSONLoader&);\n"
+        << "using NodeFactoryFn = IR::INode*(*)(JSONLoader&);\n"
```

Comment:
it looks a bit weird. After all, interface is a property of the object, there is no way to "materialize" pure interface w/o underlying object. But factory produces definite objects, not abstract interfaces, so should be `Node` here...